.github/workflows/coverage-badge.yml

```
@@ -0,0 +1,41 @@
+      with:
+        auth: ${{ secrets.GIST_TOKEN }}
+        gistID: YOUR_GIST_ID # Replace with actual gist ID
+        filename: mcp-socialmedia-coverage.json
+        label: Coverage
+        message: ${{ env.COVERAGE_PERCENTAGE }}%
+        valColorRange: ${{ env.COVERAGE_PERCENTAGE }}
+        maxColorRange: 100
+        minColorRange: 0
+
+    - name: Extract coverage percentage
+      run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2002:style:1:16: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead

(shellcheck)